drivers/vulkan/vulkan_context.cpp

```diff
@@ -359,8 +359,34 @@ Error VulkanContext::_create_physical_device() {
 		free(physical_devices);
 		ERR_FAIL_V(ERR_CANT_CREATE);
 	}
-	/* for now, just grab the first physical device */
-	gpu = physical_devices[0];
+
+	/*Find the first discrete GPU with the most VRAM*/
+	{
+		print_line("Selecting primary GPU.");
+		VkPhysicalDeviceProperties device_properties;
+		VkPhysicalDeviceMemoryProperties memory_properties;
+		gpu = physical_devices[0];
+		uint32_t largest_vram_size = 0;
+		VkPhysicalDeviceType gpu_type = VK_PHYSICAL_DEVICE_TYPE_OTHER;
+		for (uint32_t i = 0; i < gpu_count; i++) {
+			vkGetPhysicalDeviceProperties(physical_devices[i], &device_properties);
+			vkGetPhysicalDeviceMemoryProperties(physical_devices[i], &memory_properties);
+
+			/*Total all heaps in case of 3GB+1GB configurations and similar.*/
+			uint32_t memory_size = 0;
+			for (uint32_t j = 0; j < memory_properties.memoryHeapCount; j++) {
+				memory_size += memory_properties.memoryHeaps[j].size;
+			}
+
+			if ((device_properties.deviceType >= gpu_type) ||
```

Comment:
Are you sure we just want to check if DeviceType is >=? The highest is ``VK_PHYSICAL_DEVICE_TYPE_CPU``

Reply:
I don't think that will be an issue in practice because someone would have to install a software ICD. If they do that they might want it to be selected.

Reply:
IMO we should cap this at ``VK_PHYSICAL_DEVICE_TYPE_DISCRETE_GPU`` and then later add a command line option to specify which driver to use. 

My concern is that someone will have a discrete GPU and will have installed a software implementation for other reasons and will be stuck with the software version. Right now we provide no way to get around that. So until it is possible to manually force a particular driver, I think we should cap our select at dsicrete GPUs

Reply:
What about virtual gpu?

Reply:
Should we be prioritizing virtual GPUs over discrete GPUs? I'm not sure when deviceType would ever be ``VK_PHYSICAL_DEVICE_TYPE_VIRTUAL_GPU``, so I don't know what is best here. 

I feel like my above comment applies. Where a user has a discrete GPU it should be prioritized for now. Later we can add a command line argument to specify a particular GPU if users want to test software/emulated GPUs.